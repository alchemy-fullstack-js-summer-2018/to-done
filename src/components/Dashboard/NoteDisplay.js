import React, { Component } from 'react';
import PropTypes from 'prop-types';

class NoteDisplay extends Component {

  static propTypes = {
    note: PropTypes.object.isRequired
  };

  render() {
    const { note } = this.props;

    return (
      <p>
        <span>{note.title}</span>: {note.notes}, {note.completed} 
      </p>
    );
  }
}

export default NoteDisplay;