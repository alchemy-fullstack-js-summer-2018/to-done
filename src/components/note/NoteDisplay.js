import React, { Component } from 'react';
import PropTypes from 'prop-types';

class NoteDisplay extends Component {
  static propTypes = {
    note: PropTypes.object.isRequired,
    onEdit: PropTypes.func.isRequired,
    onDelete: PropTypes.func.isRequired
  };

  render() {
    const { note, onEdit, onDelete } = this.props;

    return (
      <p>
        {note.name} : {note.content}
        <button name="edit" onClick={onEdit}>Edit:</button>
        <button name="delete" onClick={onDelete}>Delete</button>
      </p>
    );
  }
}

export default NoteDisplay;