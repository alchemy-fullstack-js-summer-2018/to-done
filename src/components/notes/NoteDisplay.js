import React, { Component } from 'react';
import PropTypes from 'prop-types';

class NoteDisplay extends Component {

  static propTypes = {
    note: PropTypes.object.isRequired,
    onEdit: PropTypes.func.isRequired,
    onDelete: PropTypes.func.isRequired
  };
  
  render() {
    const { note, onEdit, onDelete } = this.props;

    return (
      <div>
        <h3>{note.title}</h3>
        <p>{note.content}</p>
        {note.completed
          ? <p>Completed!</p>
          : <p>Not completed</p>
        }
        <button name="edit" onClick={onEdit}>Edit</button>
        <button name="delete" onClick={onDelete}>delete</button>
      </div>
    );
  }
}
 
export default NoteDisplay;