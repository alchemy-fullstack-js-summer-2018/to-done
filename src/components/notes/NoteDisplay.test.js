import NoteDisplay from './NoteDisplay';
import React from 'react';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';

describe('Note display', () => {
  it('renders display, then edit', () => {
    const handleEdit = jest.fn();
    const handleDelete = jest.fn();

    const wrapper = shallow(<NoteDisplay
      note={{ title: 'title', content: 'content', completed: false }}
      onEdit={handleEdit}
      onDelete={handleDelete}
    />);

    expect(toJSON(wrapper)).toMatchSnapshot();

    wrapper.find('button[name="edit"]').simulate('click');
    const editCalls = handleEdit.mock.calls;
    expect(editCalls.length).toBe(1);
    
    wrapper.find('button[name="delete"]').simulate('click');
    const deleteCalls = handleDelete.mock.calls;
    expect(deleteCalls.length).toBe(1);
  });
});