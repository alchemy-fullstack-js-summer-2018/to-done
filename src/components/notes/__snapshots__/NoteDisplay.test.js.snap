// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Note display renders display, then edit 1`] = `
<div>
  <h3>
    title
  </h3>
  <p>
    content
  </p>
  <p>
    Not completed
  </p>
  <button
    name="edit"
    onClick={[MockFunction]}
  >
    Edit
  </button>
  <button
    name="delete"
    onClick={[MockFunction]}
  >
    delete
  </button>
</div>
`;
